openwisp_users/api/mixins.py

```diff
@@ -17,6 +18,9 @@ def _user_attr(self):
 
     organization_lookup = 'organization__in'
 
+    def get_permissions(self):
+        return super().get_permissions() + [IsAuthenticated()]
```

Comment:
what about defining it in `permission_classes`? Eg:

```python
permission_classes = [IsAuthenticated]
```

This way we can override it if needed: we may need a custom `IsAuthenticated` or anything else, in that case we can override the behavior, while with this code overriding will be harder.

What do you think?

Reply:
My idea was to make it robust so that the developers wouldn't have to worry about adding this permission class separately while creating views. But sure, we can do this instead.

For the time being, I'm keeping it simple as you said:
```python
permission_classes = [IsAuthenticated]
```

Also, I had this other idea, we could check if the user is authenticated, manually:
```python
...
if not (request.user and request.user.is_authenticated):
            raise NotFound()
...
```
Let me know if this will be better.